pytorch_lightning/loops/batch/training_batch_loop.py

```diff
@@ -263,14 +263,27 @@ def backward_fn(loss: Tensor):
             self.backward(loss, optimizer, opt_idx)
 
             # check if loss or model weights are nan
-            if self.trainer.terminate_on_nan:
+            if self.trainer.terminate_on_nan or self.trainer.detect_anomaly:
                 check_finite_loss(self.trainer.lightning_module, loss)
 
             return loss
 
         if not self._skip_backward and self.trainer.lightning_module.automatic_optimization:
             return backward_fn
 
+    def _process_closure_result(self, opt_closure_result: Optional[ClosureResult]) -> None:
```

Comment:
is this a bad merge? it looks like this has been moved out of he training batch loop

Reply:
I think so. How do I fix it?

Reply:
Hey @yopknopixx some code in the loop files have changed in recent refactors. 
1) merge master into this branch
2) remove this code here
3) update the if condition for check_finite_loss() call in loops/utilities/_process_training_step_output
4) update the if condition for detect_nan_loss() call in loops/optimizer/optimizer_loop